### PR TITLE
llvm --HEAD must require xcode

### LINKED ISSUE #4564: llvm --HEAD must require xcode
I have
```
$ xcode-select -p
/Library/Developer/CommandLineTools
```
# Please follow the general troubleshooting steps first:

- [*] Ran `brew update` and retried your prior step?
- [*] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?

### Bug reports:
```
Last 15 lines from /Users/yurikoles/Library/Logs/Homebrew/llvm/01.cmake:
  command.

CMake Warning at tools/xcode-toolchain/CMakeLists.txt:60 (message):
  Failed to detect the version of an installed copy of Xcode, falling back to
  highest supported version.  Set XCODE_VERSION to override.

CMake Error at tools/xcode-toolchain/CMakeLists.txt:80 (message):
  Could not identify toolchain dir

-- Configuring incomplete, errors occurred!
See also "/tmp/llvm-20160906-13025-1s9nevg/CMakeFiles/CMakeOutput.log".
See also "/tmp/llvm-20160906-13025-1s9nevg/CMakeFiles/CMakeError.log".
```
Running `xcode-select -s /Applications/Xcode.app` had resolved issue.

Full log: https://gist.github.com/yurikoles/e0c7a4431d65c4228f2b549fcc2a1470

### PR DESCRIPTION
**Pull request by yurikoles**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?

closes #4564

### CODE REVIEW

**Comment by yurikoles** [comment]
How to specify min Xcode version?

**Comment by zmwangx** [comment]
`depends_on :xcode => "7.3"`

Or whatever version you want here.

File: Formula/llvm.rb [hunk 1]
```diff
@@ -117,6 +117,10 @@ class Llvm < Formula
     resource "polly" do
       url "http://llvm.org/git/polly.git"
     end
+
+    if build.with? "toolchain"
+      depends_on :xcode
+    end
```
Comment (zmwangx) [inline Formula/llvm.rb]:
```ruby
depends_on :xcode => "<version>" if build.with? "toolchain"
```

**Comment by yurikoles** [comment]
How to combine Xcode version check and macOS version check? Because we need 8.0 on Sierra and up.

**Comment by zmwangx** [comment]
> How to combine Xcode version check and macOS version check?

If you want to specify hard requirements:

```ruby
depends_on :xcode => "8.0"
depends_on :macos => :sierra
```

If you want a conditional in the `install` block for instance:

```ruby
if MacOS.version == :sierra && MacOS::Xcode.installed? && MacOS::Xcode.version >= "8.0"
  # do something
end
```

**Comment by yurikoles** [comment]
@zmwangx I no longer think that we need to specify Xcode version in formulae, since there is already check for Xcode version <=8.0.

**Comment by zmwangx** [comment]
Sorry, this cannot be merged as is because at the very least you need to address my old comment. Other than that, we also need someone more knowledgeable about LLVM than me to decide if this is the right change (probably is) and whether we need additional tweaks.

**Comment by yurikoles** [comment]
I had problems building with Xcode 7.3 on Sierra, this is why I wanted to explicitly specify Xcode version. But this problem is so common that there is now fall with error if one uses old Xcode on Sierra. So I no longer see the reason to specify Xcode version in formulae.

**Comment by zmwangx** [comment]
To be clear, you need to address [this](https://github.com/Homebrew/homebrew-core/pull/4651/files#r78127572).

**Comment by yurikoles** [comment]
up & rebase

File: Formula/llvm.rb [hunk 2]
```diff
@@ -118,6 +118,9 @@ class Llvm < Formula
     resource "polly" do
       url "http://llvm.org/git/polly.git"
     end
+
+    depends_on :xcode => "8.0" if build.with? "toolchain" && MacOS.version == :sierra
+    depends_on :xcode => "7.3" if build.with? "toolchain" && MacOS.version <= :el_capitan
```
Comment (zmwangx) [inline Formula/llvm.rb]: This doesn't look right, there's no Xcode 7.3 on Yosemite and below.

**Review by zmwangx — CHANGES_REQUESTED** [review]

**Comment by yurikoles** [comment]
I will run checks on VM.

**Comment by yurikoles** [comment]
So I had tried this formula in El Capitan with Xcode 7.0 and it seems like version check doesn't work.

**Comment by MikeMcQuaid** [comment]
Passing on this due to unaddressed comments.

**Pull request closed without merging**